cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -17,177 +17,118 @@ limitations under the License.
 package kubeconfig
 
 import (
-	"bytes"
 	"crypto/x509"
 	"fmt"
-	"os"
-	"path/filepath"
 
-	"k8s.io/client-go/tools/clientcmd"
+	"crypto/rsa"
+
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	certutil "k8s.io/client-go/util/cert"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
 	"k8s.io/kubernetes/cmd/kubeadm/app/phases/certs/pkiutil"
 	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
 )
 
-// BuildConfigProperties holds some simple information about how this phase should build the KubeConfig object
-type BuildConfigProperties struct {
-	CertDir         string
-	ClientName      string
-	Organization    []string
-	APIServer       string
-	Token           string
-	MakeClientCerts bool
+// buildConfigProperties holds some simple information about how to build the KubeConfig object
+type buildConfigProperties struct {
+	ClientName   string
+	Organization []string
```

Comment:
Please make a `ClientCertAuth` struct with the unique fields there: `Organization` and `CAKey`